websites/M/MarineTraffic/presence.ts

```diff
@@ -0,0 +1,330 @@
+const presence = new Presence({
+		clientId: "1013343014711738368",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+let locationMap: string,
+	elemName: string,
+	shipStatus: string,
+	shipDest: string,
+	shipDep: string,
+	elemDesc: string;
+
+const capitalize = (s: string | never[]) =>
+	(s && s[0].toUpperCase() + s.slice(1)) || "";
+
+presence.on("UpdateData", async () => {
+	let presenceData: PresenceData = {
+		largeImageKey: "logo",
+		startTimestamp: browsingTimestamp,
+		details: "Viewing:",
+	};
+
+	const pages: Record<string, PresenceData> = {
+		"ais-historical-data": { state: "Historical AIS Data" },
+		"?asset_type=vessels": { state: "Vessels Database" },
+		"?asset_type=ports": { state: "Ports Database" },
+		"?asset_type=expected_arrivals": { state: "Expected Arrivals" },
+		"?asset_type=arrivals_departures": { state: "Port Calls" },
+		"?asset_type=lights": { state: "Lighthouses Database" },
+		"?asset_type=stations": { state: "Stations Database" },
+		"user-agreement": { state: "User Agreement" },
+		terms: { state: "Terms" },
+		"privacy-policy": { state: "Privacy Policy" },
+		"maritime-companies/search": { state: "Maritime companies search" },
+		"expand-coverage": { state: "Ship Spotting AIS Receiver" },
+		"apply-for-free-ais-receiver": { state: "Free AIS Receiver" },
+		"report-your-own-position": { state: "Report AIS Position" },
+		"online-services/plans": { state: "Online Services Plans" },
+		"photos/of/ships": { state: "Vessel photos" },
+		"ais/index/photographers": { state: "Photographers list" },
+		"users/my_account": { state: "User Account Overview" },
+		"api/account": { state: "API Services" },
+		"payment/billing": { state: "Billing Settings" },
+		"stations/index": { state: "Personal Stations" },
+		"fleets/index": { state: "User Fleets" },
+		"users/plans": { state: "Online Services & Plans" },
+		"maritime-companies/index": { state: "Business Profile" },
+		"corporate/index": { state: "Corporate Account" },
+		"users/settings": { state: "User Settings" },
+		"notifications/index": { state: "User Notifications" },
+	};
+
+	for (const [path, data] of Object.entries(pages)) {
+		if (document.location.href.includes(path))
+			presenceData = { ...presenceData, ...data };
+	}
+
+	if (document.location.hostname.startsWith("shop")) {
+		presenceData.details = "Browsing AIS e-shop:";
+		presenceData.state = document.title.split("-")[0];
+	} else if (document.location.hostname.startsWith("help")) {
+		presenceData.details = "Reading help:";
+		presenceData.state = document.title.split("–")[0];
+		presenceData.buttons = [
+			{ label: "Read this page", url: document.location.href },
+		];
+	} else if (document.location.pathname.startsWith("/blog")) {
+		presenceData.details = "Reading a blog post:";
```

Comment:
Same here right @EncryptedDev

Reply:
Same, Encrypted thumbsup
```suggestion
		presenceData.details = "Reading a blog post";
```